### PR TITLE
course details - student tab migration to MVVM

### LINKED ISSUE #318: [sub-issue] Migrate Course details page - Students tab to MVVM
**Migrate Course details page - Students tab to MVVM**

Currently, this page is developed using MVP architecture this should be migrated to MVVM

**Would you like to work on the issue?**
yes

### PR DESCRIPTION
**Pull request by gconnect**
## What this PR does
Fixes #318

**describe the purpose of this pull request and note the issue it addresses**
- This PR migrates the course details - student tab page codebase from MVP to MVVM

### CODE REVIEW

**Comment by gconnect** [comment]
@ujjwalagrawal17 please review

File: app/src/main/java/org/wikiedufoundation/wikiedudashboard/ui/coursedetail/students/view/StudentListFragment.kt [hunk 1]
```diff
@@ -42,37 +39,43 @@ class StudentListFragment : Fragment(), StudentListView {
...
 
-    override fun showProgressBar(show: Boolean) {
-        progressBar?.visibility = if (show) {
-            View.VISIBLE
-        } else {
-            View.GONE
-        }
+    private fun initializeProgressBar() {
+        studentsViewModel.progressbar.observe(this, androidx.lifecycle.Observer {
+            progressBar.visibility = if (it) View.VISIBLE else View.GONE
```
Comment (ujjwalagrawal17) [inline app/src/main/java/org/wikiedufoundation/wikiedudashboard/ui/coursedetail/students/view/StudentListFragment.kt]: You can use the ternary operator here in future. 

progressBar.visibility = it? View.VISIBLE : View.GONE

It looks more clean to me.

**Comment by ujjwalagrawal17** [comment]
Nice Job @gconnect . Thanks

File: app/src/main/java/org/wikiedufoundation/wikiedudashboard/ui/coursedetail/students/view/StudentListFragment.kt (hunk 1 above)
Comment (gconnect) [reply app/src/main/java/org/wikiedufoundation/wikiedudashboard/ui/coursedetail/students/view/StudentListFragment.kt]: Thanks will take note of it

**Pull request merged**